tests/foreman/api/test_contentview.py

```diff
@@ -1652,56 +1378,52 @@ def test_negative_readonly_user_actions(self):
         """
         user_login = gen_string('alpha')
         user_password = gen_string('alphanumeric')
-        lce = entities.LifecycleEnvironment(organization=self.org).create()
         # create a role with content views read only permissions
-        role = entities.Role().create()
         entities.Filter(
-            organization=[self.org],
+            organization=[module_org],
             permission=entities.Permission(resource_type='Katello::ContentView').search(
                 filters={'name': 'view_content_views'}
             ),
             role=role,
         ).create()
         # create environment permissions and assign it to our role
         entities.Filter(
-            organization=[self.org],
+            organization=[module_org],
             permission=entities.Permission(resource_type='Katello::KTEnvironment').search(),
             role=role,
         ).create()
         # create a user and assign the above created role
         entities.User(
-            organization=[self.org], role=[role], login=user_login, password=user_password
+            organization=[module_org], role=[role], login=user_login, password=user_password
         ).create()
-        # Create new content view entity using admin user
-        content_view = entities.ContentView(organization=self.org).create()
         cfg = get_nailgun_config()
         cfg.auth = (user_login, user_password)
         # Check that we cannot create content view due read-only permission
-        with self.assertRaises(HTTPError):
-            entities.ContentView(cfg, organization=self.org).create()
+        with pytest.raises(HTTPError):
+            entities.ContentView(cfg, organization=module_org).create()
         # Check that we can read our content view with custom user
         content_view = entities.ContentView(cfg, id=content_view.id).read()
         # Check that we cannot modify content view with custom user
-        with self.assertRaises(HTTPError):
+        with pytest.raises(HTTPError):
             entities.ContentView(cfg, id=content_view.id, name=gen_string('alpha')).update(
                 ['name']
             )
         # Check that we cannot delete content view due read-only permission
-        with self.assertRaises(HTTPError):
+        with pytest.raises(HTTPError):
```

Comment:
Optional, maybe future test enhancement:

Assert more detail of the HTTPError - should the delete result in a 400? 403? 405?  
I think its valuable test coverage to make sure the return codes make sense for the user.

Reply:
Make Sense, 

Action based on errors codes or error message.

Reply:
The `mssg` parameter in pytest.raises is available though.